### ISSUE TITLE
Singleton.m_shuttingDown causes problems for e.g. ARLocationManager when restarting the scene

### ISSUE THREAD

**Issue by TobiasWehrum** [dmbfm/unity-ar-gps-location-issues#9]
When awaking, some AR managers try to access Singleton instances of other managers. When a scene reloads, all Singleton instances have m_ShuttingDown set to true and only set that to false on Awake().

Depending on the order of Awake() calls (which is not guaranteed unless manually set in the Script Execution Order) this can lead to a situation where "[Singleton] Instance already destroyed. Returning null." is triggered.

**Comment by dmbfm** [comment]
Which version of the asset are you using? Because I believe this was modified on v3.1.0.

I think ARLocationProvider used to reference the ARLocationManager instance in the Awake, but I removed that.  And these are the only components that use `Awake`. 

But if this is still happening maybe there was a regression. Can you point out where this error appears?  
Thanks!

**Comment by TobiasWehrum** [comment]
No, I'm still on v3.0.3, so I assume this is not a regression. But if your fix was to remove referencing singleton instances in Awake(), it will still be a problem if user-written classes access your singletons in Awake() on a scene switch. (And a rather hidden one too, as it will only happen on a scene switch if the random Awake() order is unfortunate.)
